sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -192,6 +194,22 @@ def test_agglomerative_clustering():
                                                          clustering.labels_),
                             1)
 
+    # Test that using a distance matrix (affinity = 'precomputed') has same results
+    # (with connectivity constraints)
+    clustering = AgglomerativeClustering(
+            n_clusters=10,
+            connectivity=connectivity,
+            linkage="complete")
+    clustering.fit(X)
+    X_dist = squareform(pdist(X))
```

Comment:
Minor: Should probably use the scikit-learn internal `pairwise_distances` here...